### ISSUE TITLE
Paladin able to cast Blessing of Protection during Sap (on another player)

### ISSUE THREAD

**Issue by Schaka** [ccshiro/cc-buglist#1114]
Is: 
While Paladins are under the effect of Sap, they are able to cast Blessing of Protection (and possibly other blessing) on other, friendly players.

Should:
Paladin can definitely cast BOP on himself while under physical crowd control. Before 2.2.0, Paladins could cast BOP on other players while stunned. Since Sap is "incapacitation", it should probably be possible to cast BOP during Sap on other players.

"Patch 2.2.0:
- Blessing of Protection: This spell can no longer be cast on others 
  when stunned. It can only be cast on self (to break the stun) under 
  those circumstances."

**Comment by Doppelmoral** [comment]
Please try to swap weapons / shield during sap as well. Commonly possible on other servers (f.e. equipping shield during sap) but a bug

**Comment by YourNewfriend** [comment]
nah.. Paladins should at least be able to cast Blessing of Protection on themselves to break out of sap, but casting it on other players definitely sounds like a bug.

**Comment by Doope** [comment]
You can cast BoP on yourself while in physical CC, like blind and warrior fear and that works fine,but ye you shouldnt be able to cast it on others while in CC

**Comment by Shizumaru** [comment]
"other players definitely sounds like a bug". HAUAHAUHAUAHUA
 Paladin HAVE to cast BoP, HoP to a friendly target under CC....it's not a bug

**Comment by pyrokar6** [comment]
I think paladins were able to cast BoP on other players if they were blinded, can't say about when they're saped.

**Comment by Iliana-sc** [comment]
I recall this to be the correct behavior, but I don't have any proof. This should probably be investigated before it's assumed to be bugged, even if it sounds strange.

**Comment by Schaka** [comment]
This was "correct" behavior in Wrath, when they changed blessings to hands. You were able to cast HoF and HoP on other players and yourself during sap and stuns, but not other forms of CC.

It's incorrect behavior in TBC. If this had been possible during TBC; it shouldn't be hard finding video proof of a Paladin doing it, correct? I went through my small stash of downloaded videos and didn't find anything.

**Comment by Iliana-sc** [comment]
If you have evidence that it worked like this in wrath then that is a good reason to research this properly for TBC and Vanilla. There is no good reason to assume this behavior was added in wrath and didn't already exist in TBC and vanilla. I personally remember this from vanilla, though I have no evidence and I'm not willing to say I'm 100% sure.

I don't know how many videos you checked, but the assumption that it should be "easy to find a video" is flawed at best.

**Comment by Schaka** [comment]
People starting doing it in Wrath and never before. This is my personal memory of how things were in Vanilla and TBC, then changed in Wrath.
I check HammerOfGG2 and videos where people faced Paladins, like the Neilyo ones. The idea was to find something where a Paladin would cast BOP on someone else. I figured it would most likely happen in 3s or in matches like Lock/Pally vs Rogue/Druid. Unfortunately there aren't many 3s videos for TBC. Anyway, I wasn't able to find a situation where a Paladin did just that. It lines up with my memory of how things were

**Comment by AraelWindwings** [comment]
As far I remember, not a bug, the client let you cast spell which can break the CC you are under with.

The target which you cast these on does not matter. So long ago we had same question on another server, I might be wrong but we found retail video about this.

**Comment by ccshiro** [comment]
I found this patch note:

"Patch 2.2.0:
- Blessing of Protection: This spell can no longer be cast on others 
when stunned. It can only be cast on self (to break the stun) under 
those circumstances."

**Comment by Jumpka** [comment]
https://www.youtube.com/watch?v=mJT1twzaCWI#t=15m31s BoP on mage from warrior fear.
It's before the 2.2.0 patch tho (but I hope when they said "can no longer be cast on others 
when stunned" they actually meant only stuns)

**Comment by Schaka** [comment]
It would make little sense that you should be able to cast BOP while under the effect of a crowd control that renders you unable to cast spells at all. However, like I said before, they implemented it in Wrath when they changed blessings to hands (and you started being able to drop totems while silenced). I specifically remember idiot rets who would pop HOF while sapped, thinking they could break it.

Also, does warrior fear not have stun DR instead of fear DR? I vaguely remember there being something like this. That may have only been for the main target which doesn't "run away in fear".

**Comment by Jumpka** [comment]
Wait what do you mean Schaka? I just posted a TBC video where he did BoP a party member from warrior's fear. I guess I missed your point in that last comment.
Also, yes, warrior fear is in controlled stuns DR group, but what does it have to do with this?

**Comment by Schaka** [comment]
Because that means that warrior fear might be treated as a stun by the game, leading to the paladin being able to cast BOP under its effect pre-2.2. I assumed that being able to cast it on others while stunned was a faulty mechanic (hence the fix, or change).
Because like I said, I don't remember being able to cast it while under the effect of crowd control that renders you unable to cast spells. 

There are some odd mechanics in the game. For example, you can't even use Divine Shield while cycloned.

**Comment by Jumpka** [comment]
Schaka I'm pretty sure when we are talking about patch notes and stuff like that, unless they say something like "controlled stun DR group" or so, referring to stun actually means stuns, and not things that are in stun DR (warrior's fear is treated like a fear expect for the DR group), would that mean that we can still cast it for example from blind, since that has nothing to do with stuns? And if they meant warrior's fear, blind and sap, why not simply write "the spell can no longer be cast on others when under effect of CC. It can only be cast on self (to break the CC) under those circumstances." ?

**Comment by Schaka** [comment]
Because I'm assuming it already worked that way and only stuns were excluded from it. I'm saying that it's a possibility warrior fear, despite the fear effect, may have been considered a stun, which is why it worked pre-2.2.

I'm also saying there's a pretty good chance you're right.

**Comment by Jumpka** [comment]
The thing that you are then saying is, pre 2.2.0, you could BoP yourself into Warr fear / stuns / blind / sap, but could only BoP others while in Warr fear / stuns and not blind and sap. Then 2.2 came, they removed any casting of BoP while CCed, but in the patch notes said they removed stuns from the party cast part, instead of just saying you can't BoP others while in CC. Seems pretty strange tbh

**Comment by Schaka** [comment]
I'm saying pre-2.2, you could bop yourself in stuns and warrior fear, possibly blind, but not other CC - because casting during a crowd control effect is usually not possible. Pretty sure the polymorph spell even specifically mentions that.

With 2.2, they removed it from stuns - possibly like you're saying from all CC (but I doubt it). I'm saying it's a possibility that pre-2.2 you could ONLY cast it during stuns and warrior fear may have been an exception (like its DR is an exception) to that rule.

**Comment by Jumpka** [comment]
Wait now I'm completelly lost, I would never say polymorph, I also didn't say all CCs myself, I was just trying to rephrase YOUR point to show how strange/illogical it is. Also are you telling me that you couldn't BoP yourself in sap before 2.2??

**Comment by Schaka** [comment]
I know you were just quoting me. And I quoted you on that. That's what the "all CC" part was about. I don't think my point is illogical, because it lines up with the memories I have of that time. I'm trying to make sense of them, given the information at hand.

Yes, you couldn't bop yourself in sap before 2.2. and you couldn't do so after. I am 99% sure of that, because it never happened. If it were possible, paladins would've frequently done so in the opener. Because a 2minute BOP CD and 1minute bubble CD on themselves while trinket was still up wouldn't have been fatal (not in 2s anyway). I thought I could find some proof of that by looking at actionbars on videos, but during most CC, all spells look available on the action bars and just give an error message. I have found that blind post-2.2 only shows trinket, BOP and divine shield as available though. 

So now I am unsure whether blind should be breakable on yourself with BOP but Sap shouldn't.

Edit:I just saw Feel the Justice 3 and at 11:44, you can see him being feared and unable to cast BOP. So I think it has something to do with the physical effect of it. So you can probably break blind and definitely break warrior fear, but you likely cannot break sap (or BOP others during sap), because it shows all your actionbars as active (just like stuns do) and then gives an error message when trying to BOP.

**Comment by Jumpka** [comment]
Schaka "I just saw Feel the Justice 3 and at 11:44, you can see him being feared and unable to cast BOP. So I think it has something to do with the physical effect of it." and again I am completely lost, what did you expect, him being able to cast it from MAGICAL fear? I thought that the part about it being  castable just from physical CCs and nothing else was the part everyone knew about.

**Comment by Jumpka** [comment]
Oh and since you mentioned Feel the justice 3 and also said "Yes, you couldn't bop yourself in sap before 2.2. and you couldn't do so after. + So now I am unsure whether blind should be breakable on yourself with BOP but Sap shouldn't." in the same comment, all I can say is: https://www.youtube.com/watch?v=OveRM0T6_ls#t=6m25s
Nice
meme

**Comment by Schaka** [comment]
So you can BOP yourself. It's probably safe to say, that you could also BOP others under the effect of sap then.

**Comment by Jumpka** [comment]
OP could you edit the things that got discussed here into the first post so shiro/nim doesn't have to go trough 25 comments?
